configure.in

```
@@ -261,11 +265,33 @@ if test "$HAS_TABLEAM" == 'yes'; then
 
 fi # test "$HAS_TABLEAM" == 'yes'
 
+
+PGAC_ARG_BOOL(with, security-flags, yes,
+              [do not use security flags])
+AC_SUBST(with_security_flags)
+
+if test "$with_security_flags" = yes; then
+# Flags taken from: https://liquid.microsoft.com/Web/Object/Read/ms.security/Requirements/Microsoft.Security.SystemsADM.10203#guide
+
+# We always want to have some compiler flags for security concerns.
+SECURITY_CFLAGS="-fstack-protector-strong -Wformat -Wformat-security -Werror=format-security -O2 -fpic -shared -D_FORTIFY_SOURCE=2 -D_FORTIFY_SOURCE=2 -z noexecstack -fpic -shared -Wl,-z,relro -Wl,-z,now"
+CITUS_CFLAGS="$CITUS_CFLAGS $SECURITY_CFLAGS"
+AC_MSG_NOTICE([Blindly added security flags for linker: $SECURITY_CFLAGS])
+
+# We always want to have some clang flags for security concerns.
+SECURITY_BITCODE_CFLAGS="-fsanitize=safe-stack -fstack-protector-strong -flto -fvisibility=hidden -fPIC -Wformat -Wformat-security -Werror=format-security"
```

This is the flag I see on the liquid page

```suggestion
SECURITY_BITCODE_CFLAGS="-fsanitize=safe-stack -fstack-protector-strong -flto -fsanitize=cfi -fPIC -Wformat -Wformat-security -Werror=format-security"
```

---

With this flag, I can not build on Debian Buster. I think we should leave it as is